docs/references/security/rust-canister-development-security-best-practices.md

```diff
@@ -183,43 +183,157 @@ A canister could be rendered unusable so it could never be upgraded again e.g. d
 
 ## Inter-Canister Calls and Rollbacks
 
-### Don’t panic after await and don’t lock shared resources across await boundaries
+### Message Execution Basics
+
+To understand the issues around async inter-canister calls, one needs to understand a few properties about message execution. This is also explained in the [community conversation on Security Best Practices](https://www.youtube.com/watch?v=PneRzDmf_Xw&list=PLuhDt1vhGcrez-f3I0_hvbwGZHZzkZ7Ng&index=2&t=4s). A _message_ is a set of consecutive instructions that a subnet executes for a canister. The following properties are essential: 
+
+**Property 1**: Only a single message is processed at a time. So message execution is sequential, and never parallel.
+
+**Property 2**: Each call (query / update) triggers a message. When an inter-canister call is made, the code after the call (the callback) is executed as a separate message. 
+
+For example, consider the following code in Motoko: 
```

Comment:
```suggestion
For example, consider the following Motoko code: 
```

Reply:
It's a bit weird that this is a "Rust Canister Development" but we have sample Motoko code